src/videojs-hls.js

```diff
@@ -1030,6 +1030,11 @@ videojs.Hls.translateMediaIndex = function(mediaIndex, original, update) {
     // let the media index be zero when there are no segments defined
     return 0;
   }
+  
+  if (original.mediaSequence === update.mediaSequence) {
+	// playlist sequence didn't change
+	return mediaIndex;
```

Comment:
Yes, the problem is when `translatedMediaIndex === update.segments.length`. The index gets increased and falls out of the playlist.